### DISCUSSION TITLE
Are you able to restrict the visibility of slash / application commands based on permission level within the bot itself?

### DISCUSSION THREAD

**Discussion by Immortql** [Rapptz/discord.py#9859] [Q&A] [answered]
Basically as the title states. Wondering if there's a way to essentially hide slash commands from users if they lack the required permissions to run them. I know you can do it within the Discord integration tab if you have admin / owner rights- but I'll explain more about why I'm asking this down a bit further.

I know you can use decorators where the command functions are defined to prevent execution like so. 

```python
# Only let the command run if the user has the `ban_members` permission
@commands.has_permissions(ban_members=True)
```

But that doesn't do anything about their visibility.

I recently invited another bot to my server (Kettu) and it does this automatically without any need for input on my end.
As the owner, I can see moderation commands such as mute, kick, ban, etc.
But normal users (tested on my alternate account), cannot see said commands. This is being done without intervention on the integrations tab.

So I was wondering if this behavior is possible to replicate in the latest version of discord.py.

(I am aware that Kettu uses a custom fork of discord.js, and not all features may have parity between the two. But that's why I thought it'd be worth it to ask)

**Comment by Zalk0** [comment, marked as answer]
Hey,
I think you're looking for this https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=default_permissions#discord.app_commands.default_permissions
```py
@app_commands.default_permissions(ban_members=True)
```
This decorator changes the default permission needed for seeing/using the command Discord side whereas has_permissions does a check on the bot side to be sure the user running the command has the permissions.

**Reply by Immortql** [reply]
Yep, that's perfect. Thank you!
